src/main/native/GCM.c

```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
-
-		fc = (keyLen == 16) ? 18 : (keyLen == 24) ? 19 : 20;
-
-		memset(&hashSubkey,0, blockSize);
-		memset(&zeros,0, blockSize);
-		memcpy(&hashSubkeyParamBlock, key, keyLen);
-		z_km_native((signed char*) &zeros,blockSize,0,(signed char*) &hashSubkey,0,(signed char*) &hashSubkeyParamBlock,fc);
-
-		// Computing GHash for IV
-		ghashParamBlockPtr = (char*) &ghashParamBlock;
-		memset(&ghashParamBlock,0, blockSize);
-		memcpy(ghashParamBlockPtr + blockSize, &hashSubkey, blockSize);
-
-		if (ivLength >= blockSize) {
-			lenn = ivLength - (ivLength % blockSize);
-			z_kimd_native((signed char *) iv, lenn, offset, (signed char *) &ghashParamBlock, 65);
-			ivLength -= lenn;
-			offset += lenn;
-		}
-		
-		if (ivLength > 0) lastIVLen *= 2;
-		char lastIV[lastIVLen];
-		memset(&lastIV,0,lastIVLen);
-		if (ivLength > 0) memcpy(&lastIV, iv + offset, ivLength);
-
-		// Appending IV.length
-		putLongtoByteArray(ivLengthOG * 8, (signed char *) &lastIV, lastIVLen - 8);
-		z_kimd_native((signed char *) &lastIV, lastIVLen, 0, (signed char *) &ghashParamBlock, 65);
-
-		// Updating addedParam
-		for (i = 0; i < blockSize; i++) addedParams[J0Offset + i] = ghashParamBlock[i];
-	}
-	#endif
-}
-
-int checkTagMismatch(char* input, int inputLen, char* parm_block, int tagOffset, int tagLen) {
-	#if defined (S390_PLATFORM) || defined(__MVS__)
-	// check entire authentication tag for time-consistency
-
-	int i = 0;
-	int mismatch = 0;
-
-#if 0
-	char tag[tagLen];
-	char newTag[tagLen];
-
-	memcpy(&tag, input + inputLen, tagLen);
-	memcpy(&newTag, parm_block + tagOffset, tagLen);
-#else
-    char *tag = input + inputLen;
-    char *newTag = parm_block + tagOffset;
-#endif
-
-	for (i = 0; i < tagLen; i++) mismatch |= tag[i] ^ newTag[i];
-	
-	return (mismatch == 0) ? 0 : -1;
-#else
-	return 0;
-#endif
-}
 /*============================================================================
+*
 * Class:     com_ibm_crypto_plus_provider_ock_NativeInterface
 * Method:    internal functions
-* Signature: 
+* Signature: (JJ)V
 */
-int GCM_decrypt_core (JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+int GCM_InitForUpdateDecrypt_core (JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
 	unsigned char * key, int keyLen,
 	unsigned char * iv,  int ivLen,
-	unsigned char * ciphertext, int ciphertextOffset, int ciphertextLen,
-	unsigned char * plaintext, int plaintextOffset,
-	unsigned char * aad, int aadLen, int tagLen) {
+	unsigned char * aad, int aadLen, unsigned long *updateOutlen) {
+#if defined(AIX) || defined(WINDOWS) || defined(MAC) || defined (LINUX) || defined(__MVS__)
 
-	long unsigned updateOutlen   = 0;
-	long unsigned updateAADlen   = 0;
-	long unsigned finalOutlen    = 0;
+	 long unsigned updateAADlen   = 0;
+	 /* long unsigned finalOutlen    = 0; */
 	int           rc             = ICC_OSSL_SUCCESS;
-	static const char * functionName = "NativeInterface.GCM_decrypt_core";
+	
+	static const char * functionName = "NativeInterface.GCM_InitForUpdateDecrypt_core";
```

Comment:
Should we leave a space after the `*` too?

Reply:
I think we should have no space after the * only before them.